api/src/services/graphql/resolvers/system.ts

```diff
@@ -222,57 +213,9 @@ export function injectSystemResolvers(
 		},
 	});
 
-	const Collection = schemaComposer.createObjectTC({
-		name: 'directus_collections',
-	});
-
-	const Field = schemaComposer.createObjectTC({
-		name: 'directus_fields',
-	});
-
-	const Relation = schemaComposer.createObjectTC({
-		name: 'directus_relations',
-	});
-
-	const Extension = schemaComposer.createObjectTC({
-		name: 'directus_extensions',
-	});
-
-	const composers: BaseTypeComposers = {
-		Collection,
-		Field,
-		Relation,
-		Extension,
-	};
+	const Collection = getCollectionType(schemaComposer, schema, 'read');
```

Comment:
Now that we do not pass these objects into `resolveSystemAdmin` we should be able to move them into the read access check. This will provide a small perf improvement when access is denied

Reply:
Refactored via f64620818be658325a73a56c8ba6efdd4246932d